Safe multi-tenancy in case contacts report

Thrillberg opened the pull request:
### What github issue is this PR for, if any?
Resolves #1340

### What changed, and why?
I've added a `casa_org` scope to `CaseContact`, with the hopes that we can filter out all `CaseContact`s from other CASA organizations. Unfortunately, it's not working yet (but I wanted to get this wip up in case anyone can give feedback or take over if I run out of time).

### How will this affect user permissions?
I think it shouldn't affect permissions

### How is this tested? (please write tests!) 💖💪
I wrote a test that is now passing, but it's broken when I test it in development. Anyone have any ideas why or what I'm doing wrong?

### Screenshots please :)


### Feelings gif (optional)

![ambivalent dog with flag](https://media.giphy.com/media/3o7aCSX8FQQRYJYPv2/source.gif)

littleforest commented on app/models/case_contact_report.rb:
```diff
@@ -9,6 +9,7 @@ def filtered_case_contacts(args)
     CaseContact
       .supervisors(args[:supervisor_ids])
       .creators(args[:creator_ids])
+      .casa_org(args[:casa_org_id])
```
@Thrillberg thanks for taking a look at this! We actually don't want to be able to pass in a different casa org (an admin for a particular casa org shouldn't ever be able to get info for a different organization). Instead I think you want to take your casa_org scope and apply it, something like:

```
CaseContact
  .joins(:casa_case).where(casa_cases: { casa_org: current_organization })
  .supervisors(...)
  .creators(...)
 ...
```

Thrillberg replied:
Thanks for your help on this @littleforest! I'm a little confused about how the `CaseContactReport` would have access to the `current_organization` variable. My understanding was that `current_organization` is available to controllers and from the controller would have to be passed into any collaborators like this.

If you look at the `CaseContactReportsController` change I made, the `casa_org_id` is always the current organization's id and is not changeable from the front end. (That is the intent of my code anyway. Whether it actually works that way may be questionable! 😆)

littleforest replied:
Whoops, @Thrillberg you are totally right! Great. As long as the `casa_org_id` from the front-end is not changeable (or hackable) then all is good.

compwron commented:
This looks like it works! Does it not work?
You have a brand new spec merge conflict because I just merged another PR that touched nearby unrelated code, sorry. 
Thank you for working on this! 💪✨

Thrillberg committed:
Safe multi-tenancy in case contacts report

This change adds a casa_org_id to the report parameters in order to
filter CaseContacts by CASA organization. Under no circumstances should
data from one CASA organization leak into reports for another.

Fixes #1340

Thrillberg commented:
@compwron Here's a screenshot of the report I get when I'm signed in as a CASA 2 admin. It's worth noting that in the report preparation screen, CASA org 1 volunteers and supervisors were listed in the form. So the multitenancy is broken elsewhere, too, I think.
<img width="755" alt="Screen Shot 2020-12-05 at 7 12 38 PM" src="https://user-images.githubusercontent.com/10481391/101268131-d9ad4780-372d-11eb-91a5-4b6b7f0500c6.png">
I've gone ahead and updated this branch against `main` so that if we want to merge it, it's ready to go!

compwron reviewed (approved):
Oy. Well, it's an improvement at least... we will do this! :)
